docs/nodemcu-lrm.md

```diff
@@ -0,0 +1,331 @@
+
+Any caught error is posted to a separate NodeMCU task which reports the error.  The error reporter is defined in the registry entry "`onerror`".  The default action is to print the error and then set a 1 sec one-shot timer to restart the CPU.  (One second is enough time to allow the error to be sent over the network if redirection to a telnet session is in place.)  If the `onerror` entry is set to `print` for example, then the error is simply printed without restarting the CPU.
+
+Note that the Lua runtime does not call the error handler if the error is an out-of memory one, so in this case the out-of-memory error is posted to the error reporter without a traceback.
+
+#### luaL_posttask
+
+`  int luaL_posttask (lua_State* L, int prio);`         [-1, +0, e]
+
+Posts a task to execute the function popped from the stack at the specified user task priority
+-  `prio` one of:
+   -  `LUA_TASK_LOW`
+   -  `LUA_TASK_MEDIUM`
+   -  `LUA_TASK_HIGH`
+
+Note that the function is invoked with the priority as its parameter.
+
+#### luaL_pushlfsmodule
+
+`  int luaL_pushlfsmodule ((lua_State *L);`         [-1, +1, -]
+
+This function pops a module name from the stack. If this is a string, LFS is loaded and it contains the named module then its closure is pushed onto the stack as a function value, otherwise `nil` is pushed.
+
+Returns the type of the pushed value.
+
+#### luaL_pushlfsmodules
+
+`  int luaL_pushlfsmodules (lua_State *L);`         [-0, +1, m]
+
+If LFS is loaded then an array of the names of all of the modules in LFS is pushed onto the stack as a function value, otherwise `nil` is pushed.
+
+Returns the type of the pushed value.
+
+#### luaL_pushlfsdts
+
+`  int luaL_pushlfsdts (lua_State *L);`         [-0, +1, m-]
+
+If LFS is loaded then the Unix-style date-timestamp for the compile time of the image is pushed onto the stack as a integer value, otherwise `nil` is pushed.  Note that the primary use of this stamp is to act as a unique identifier for the image version.
+
+Returns the type of the pushed value.
+
+#### luaL_reref
+`  void (luaL_reref) (lua_State *L, int t, int *ref);`         [-1, +0, m]
+
+Variant of [luaL_ref](https://www.lua.org/manual/5.3/manual.html#luaL_ref).  If `*ref` is a valid reference in the table at index t, then this is replaced by the object at the top of the stack (and pops the object), otherwise it creates and returns a new reference using the `luaL_ref` algorithm.
+
+#### luaL_rometatable
+
+`  int (luaL_rometatable) (lua_State *L, const char* tname, const ROTable *p);`         [-0, +1, e]
+
+Equivalent to `luaL_newmetatable()` for ROTable metatables.  Adds key / ROTable entry to the registry  `[tname] = p`, rather than using a new RAM table.
+
+#### luaL_unref2
+
+`  luaL_unref2(l,t,r)`
+
+This macro executes `luaL_unref(L, t, r)` and then assigns `r = LUA_NOREF`.
```

Comment:
Handy. Can I assume it's got a `do { luaL_unref(L, t, r); r = LUA_NOREF; } while (0)` safe-guard so it's safe to use as
```
if (something)
  luaL_unref2(L, t, r);
```
?

Reply:
First confession time: no I missed this so 15-all on the C syntax competition.  So **fixed** this.

I did a quick grep-fest of the headers and also came up with:
-  4 cases of trailing `;` which shouldn't be there. **Fixed**.
-  only one `do { } while(0)` use.  My macro (because single prec fmod is missing form the xtensa lib):
```C
#define luai_nummod(L,a,b,m) do { \
  if (b==0) luaG_runerror(L,"modulo by zero"); \
  (m) = (a) - floor((a)/(b))*(b); \
  if ((m)*(b) < 0) (m) += (b); \
} while (0)
```
-  2 else stealers
-  8 bare `{ }` constructs, such as `#define lua_tmpnam(b,e)	{ e = (tmpnam(b) == NULL); }` 

The Lua headers just use the bare brace constructs and avoid the  `do { } while(0)` even though they aren't syntactically identical.  I don't think that I need to change bare brace constructs in the private headers.